docs/misc/build-dependencies.md

```diff
@@ -0,0 +1,199 @@
+# Ostree Build Dependencies
```

Comment:
Do we need this document? One shouldn't encounter this if using `dnf builddep ostree`, right?
Though it might be worth linking to the CI scripts for e.g. test dependencies.

Reply:
I've run into some of the missing deps covered below, but I think it was because I never used `dnf -y install @buildsys-build`;  that seems to land most of the additional deps not covered by `dnf -y build-dep ostree`

Reply:
@Bubblemelon or @rfairley can you confirm that this file is no longer necessary if one does `dnf install -y @buildsys && dnf builddep -y ostree` as instructed in the other document?